0101-nix formatting.md

```diff
@@ -0,0 +1,65 @@
+---
+feature: nix_formatting
+start-date: 2021-08-17
+author: Raphael Megzari
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+
+[summary]: #summary
+
+Decide on a recommended automated formatter for nix files in nixpkgs.
+
+# Motivation
+
+[motivation]: #motivation
+
+Prevent debate around how things should be formatted.
+
+# Detailed design
+
+[design]: #detailed-design
+
+The implementation of this RFC should include several parts
+
+- Agree on an automated formatter. [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt) is proposed here.
+- If adopted, agree on a schedule to format nixpkgs codebase. Should it be done at once, or on a per package basis.
+- Potentially agree on a hook to enforce formatting
```

Comment:
This could be implemented as a github action pretty easily. `$(nix-build -A nixpgks-fmt)/bin/nixpkgs-fmt .`

Reply:
There is also [treefmt](https://github.com/numtide/treefmt), which might be used to exclude specific path to address problematic files.
Here is a test run of treefmt on [nixpkgs](https://github.com/0x4A6F/nixpkgs/tree/master-fmt).

List to consider for exclusion:
- [rPackages](https://github.com/NixOS/rfcs/pull/101#issuecomment-929717228)